utils/numcheck/calcelastic

```diff
@@ -0,0 +1,215 @@
+        binary, disp, coords, latvecs, specienames, species, origin)
+
+    print_elastic(elastic)
+
+
+def parse_arguments():
+    """Parses command line arguments"""
+
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+
+    msg = "Specify the displacement of the atoms (unit: ANGSTROM)"
+    parser.add_argument("-d", "--displacement", type=float, dest="disp",
+                        default=1e-5, help=msg)
+
+    msg = "DFTB+ binary"
+    parser.add_argument("binary", help=msg)
+
+    args = parser.parse_args()
+    return args
+
+
+def calculate_elastic(binary, disp, coords, latvecs, specienames, species,
+                        origin):
+    """Calculates elastic constants by finite differences"""
+
+    # to hold stress tensor
+    stress = np.empty((6, 2), dtype=float)
+    # to hold resulting elastic tensor
+    elastic = np.empty((6,6), dtype=float)
+
+    # voight convention for 2 index to 1 of tensors
+    voight = [[0,0],[1,1],[2,2],[1,2],[0,2],[0,1]]
+
+    for ii in range(6):
+        for kk in range(2):
+            strain = np.zeros((3,3), dtype = float)
+            for jj in range (3) :
+                strain[jj][jj] = 1.0
+            strain[voight[ii][0]][voight[ii][1]] += 0.5*float(2 * kk - 1) * disp
+            strain[voight[ii][1]][voight[ii][0]] += 0.5*float(2 * kk - 1) * disp
+            newcoords = np.array(coords)
+            newcoords = cart2frac(latvecs,newcoords)
+            newvecs = np.array(latvecs)
+            newvecs = np.dot(newvecs,strain)
+            newcoords = frac2cart(newvecs,newcoords)
+            writegen("geo.gen", (specienames, species, newcoords, origin, newvecs))
+            subprocess.call([binary,])
+            fp = open("autotest.tag", "r")
+            txt = fp.read()
+            fp.close()
+            match = STRESS_PATTERN.search(txt)
+            if match:
+                stress[0,kk] = float(match.group("value11"))
+                stress[1,kk] = float(match.group("value22"))
+                stress[2,kk] = float(match.group("value33"))
+                stress[3,kk] = 0.5*float(match.group("value23"))
+                stress[3,kk] += 0.5*float(match.group("value32"))
+                stress[4,kk] = 0.5*float(match.group("value13"))
+                stress[4,kk] += 0.5*float(match.group("value31"))
+                stress[5,kk] = 0.5*float(match.group("value12"))
+                stress[5,kk] += 0.5*float(match.group("value21"))
+            else:
+                raise "No match found!"
+            for jj in range(6):
+                elastic[ii][jj] = (stress[jj][0] - stress[jj][1]) / (2.0 * disp)
+
+    return elastic
+
+
+def print_elastic(elastic):
+    """Prints calculated elastic constants."""
+
+    for ii in range(6):
+        for jj in range(6):
+            elastic[ii][jj] /= (PRESURE_AU * 1.0E9) # Convert to GPa
+    print("\nElastic constants (GPa):")
+    for ii in range(6):
+        print (("%12.4f"*6) % tuple(elastic[ii]))
+
+
```

Comment:
I guess, the routines below are just copy-paste versions of the one in `calcderivs`. Why not putting them into a Python module in this directory (e.g. `numchekhelpers`) and letting both programs import it?